engine/src/test/java/org/camunda/bpm/engine/test/bpmn/tasklistener/TaskListenerEventLifecycleTest.java

```diff
@@ -0,0 +1,586 @@
+                                                                                  CompletingTaskListener.class);
+    testRule.deploy(model);
+    runtimeService.startProcessInstanceByKey("process");
+    Task task = taskService.createTaskQuery().singleResult();
+
+    // when
+    taskService.complete(task.getId());
+
+    // then
+    LinkedList<String> orderedEvents = RecorderTaskListener.getOrderedEvents();
+
+    assertThat(orderedEvents.size()).isEqualTo(1);
+    assertThat(orderedEvents.getFirst()).isEqualToIgnoringCase(TaskListener.EVENTNAME_CREATE);
+  }
+
+  // DELETE phase
+
+  @Test
+  public void shouldFireDeleteEventLastWhenProcessDeleted() {
+    // given
+    createAndDeployModelWithTaskEventsRecorderOnUserTask(TRACKED_EVENTS);
+    ProcessInstance processInstance = runtimeService.startProcessInstanceByKey("process");
+
+    // when
+    runtimeService.deleteProcessInstance(processInstance.getId(), "Canceled!");
+
+    // then
+    LinkedList<String> orderedEvents = RecorderTaskListener.getOrderedEvents();
+
+    assertThat(orderedEvents.size()).isEqualTo(2);
+    assertThat(orderedEvents.getFirst()).isEqualToIgnoringCase(TaskListener.EVENTNAME_CREATE);
+    assertThat(orderedEvents.getLast()).isEqualToIgnoringCase(TaskListener.EVENTNAME_DELETE);
+  }
+
+  @Test
+  public void shouldNotFireUpdateEventOnPropertyChangesInDeleteListener() {
+    // given
+    BpmnModelInstance model = createModelWithTaskEventsRecorderOnAssignedUserTask(TRACKED_EVENTS,
+                                                                                  null,
+                                                                                  TaskListener.EVENTNAME_DELETE,
+                                                                                  CandidateUserAssignment.class);
+    testRule.deploy(model);
+    ProcessInstance processInstance = runtimeService.startProcessInstanceByKey("process");
+
+    // when
+    runtimeService.deleteProcessInstance(processInstance.getId(), "Canceled!");
+
+    // then
+    LinkedList<String> orderedEvents = RecorderTaskListener.getOrderedEvents();
+
+    assertThat(orderedEvents.size()).isEqualTo(2);
+    assertThat(orderedEvents.getFirst()).isEqualToIgnoringCase(TaskListener.EVENTNAME_CREATE);
+    assertThat(orderedEvents.getLast()).isEqualToIgnoringCase(TaskListener.EVENTNAME_DELETE);
+  }
+
+  @Test
+  public void shouldNotFireAssignmentEventOnAssigneeChangesInDeleteListener() {
+    // given
+    BpmnModelInstance model = createModelWithTaskEventsRecorderOnAssignedUserTask(TRACKED_EVENTS,
+                                                                                  null,
+                                                                                  TaskListener.EVENTNAME_DELETE,
+                                                                                  CandidateUserAssignment.class);
```

Comment:
I suspect you wanted to use the listener that assigns the task, right?
```suggestion
                                                                                  AssigneeAssignment.class);
```

Reply:
Yes, nice catch! Thanks.